packages/api/src/api/ProviderApi.ts

```diff
@@ -79,22 +79,51 @@ export default class ProviderApi extends HttpClientBase implements ProviderApi {
         userAccount: AccountId,
         blockNumber: number,
         commitmentId?: string,
-        maxVerifiedTime?: number
+        maxVerifiedTime?: number,
+        dappUserSignature?: string
     ): Promise<ImageVerificationResponse> {
         const payload: {
             [ApiParams.dapp]: AccountId
             [ApiParams.user]: AccountId
             [ApiParams.blockNumber]: number
+            [ApiParams.dappUserSignature]?: string
             [ApiParams.commitmentId]?: string
             [ApiParams.maxVerifiedTime]?: number
-        } = { dapp: dapp, user: userAccount, blockNumber }
+        } = { dapp: dapp, user: userAccount, blockNumber, dappUserSignature }
         if (commitmentId) {
             payload['commitmentId'] = commitmentId
         }
         if (maxVerifiedTime) {
             payload['maxVerifiedTime'] = maxVerifiedTime
         }
-        return this.post(ApiPaths.VerifyCaptchaSolution, payload as VerifySolutionBodyType)
+
+        return this.post(ApiPaths.VerifyCaptchaSolutionDapp, payload as VerifySolutionBodyType)
+    }
+
+    public verifyUser(
+        dapp: AccountId,
+        userAccount: AccountId,
+        blockNumber: number,
+        commitmentId?: string,
+        maxVerifiedTime?: number,
+        dappUserSignature?: string
+    ): Promise<ImageVerificationResponse> {
+        const payload: {
```

Comment:
``` 

const payload: VerifySolutionBodyType = {
        [ApiParams.dapp]: dapp,
        [ApiParams.user]: userAccount,
        [ApiParams.blockNumber]: blockNumber,
    
etc ...
```

Would be more simple to do something like ^ here. Will probably resolve below type casting

Reply:
I have done